aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -997,9 +996,18 @@ Tensor &any_out(Tensor &result, const Tensor &self, int64_t dim, bool keepdim) {
 }
 
 Tensor &amin_out(Tensor& result, const Tensor& self, IntArrayRef dim, bool keepdim) {
-  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:", self.scalar_type(), result.scalar_type());
+  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:",
+              self.scalar_type(), result.scalar_type());
+  auto sizes = self.sizes();
+  if (sizes.size() != 0) {
```

Comment:
Fixed with `ndimension`. The `wrap_dim` function ensures dims are not out of bounds for scalars.

Reply:
No, the new code seems functionally equivalent to the old code.

Let me explain in more detail. Suppose I call this function with `amin(torch.zeros({}), (1, 3))`. In your current code, you skip the dim test entirely, because `torch.zeros({}).ndimension() == 0`. But 1 and 3 are out of bounds for a 0-dim tensor, and we should raise errors for them.

Reply:
`wrap_dim` doesn't have anything to do with it; as you can see in the implementation

```
static inline int64_t maybe_wrap_dim(int64_t dim, int64_t dim_post_expr, bool wrap_scalar=true) {
  if (dim_post_expr <= 0) {
    if (!wrap_scalar) {
      TORCH_CHECK_INDEX(false, "dimension specified as ", dim, " but tensor has no dimensions");
    }
    dim_post_expr = 1; // this will make range [-1, 0]
  }

  int64_t min = -dim_post_expr;
  int64_t max = dim_post_expr - 1;
  if (dim < min || dim > max) {
    TORCH_CHECK_INDEX(false,
      "Dimension out of range (expected to be in range of [",
      min, ", ", max, "], but got ", dim, ")");
  }
  if (dim < 0) dim += dim_post_expr;
  return dim;
}
```

dimension can still be out of range even with wrapping.